models/public/mozilla-deepspeech-0.8.2/pre-convert.py

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/env python3
+#
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+#  This script is automatically executed by open_model_zoo/tools/downloader/converter.py,
+#  and runs pbmm_to_pb.py and scorer_to_kenlm.py.
+#
+import sys
+import argparse
+import subprocess
+
+from pathlib import Path
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input_dir', type=Path)
+    parser.add_argument('output_dir', type=Path)
+    args = parser.parse_args()
+
+    subprocess.run([
+        sys.executable, '--',
+        str(Path(sys.argv[0]).with_name('pbmm_to_pb.py')), '--overwrite', '--',
```

Comment:
It seems unnecessary for these scripts to be scripts. Why not import them as modules and just call functions in them?

Reply:
1) Because I was given yolo-v4-tf as an example of pre-comvert.py script, and it call Python scripts this way. 2) Because they expose command line interface. If we call functions, we'll depend on internal features of their implementations. I wanted them to have command line interfaces for the users who want to convert their own models.
OK? Or you insist on calling functions?

Reply:
> Because I was given yolo-v4-tf as an example of pre-comvert.py script, and it call Python scripts this way.

FWIW, the reason it does that is because those scripts were written by someone else, and intended to be run as scripts. You did not have this constraint, as you were writing the conversion code yourself.

> I wanted them to have command line interfaces for the users who want to convert their own models.

I guess that's fine (although I would say that users who want to convert their own models should just save them in a supported format to start with).